File: IGI/LR3/task1/taylor.py
from .gener import gener
import math

def taylor(x:float, eps:float) -> tuple:
    """Counting value of taylor."""
    mathF = math.asin(x)
    F:float = x
    maxN: int = 500
    n: int = 0
    prev: float = x
    for i in gener(prev, x) :
        n+=1
        F+=i
        if (abs(F - mathF) < eps) or (n > maxN):
            break
    return n, F, mathF
        

File: STRWEB/LR3/src/hotelapp/static/js/taylor.js
function mathF(x) {
  return Math.asin(x);
}

function taylor(x, n) {
  let y = x;
  let i = 0;
  let prev = x;
  for (1 = 1; i < n; i++) {
    prev *= (x * x * (2 * i - 1) * 2 * i) / (4 * i * i * (2 * i + 1));
    y += prev;             
  }
  return y;
}

